core/search/templates/search/search_results.js

```diff
@@ -7,12 +7,24 @@ function collapseResults( group ) {
     }
     
     var results = group.find( 'ol' ).children( 'li' );
-    if ( results.length > 5 && dataModel !== openGroup ) {
-        results.hide().slice( 0, 5 ).show();
-        if (results.length > 50 && dataModel != 'Staff Directory') {
-            var numMore = 45;
+    if ( dataModel !== openGroup && (
+         ( dataModel == 'Staff Directory' && results.length > 20 ) ||
+         ( dataModel != 'Staff Directory' && results.length > 5 )
+    ) ) {
```

Comment:
This conditional is pretty ugly. I'm very open to suggestions for how to do it more efficiently or more readably.